### ISSUE TITLE
Radio values disappear upon dragging repeating sortable field group

### ISSUE THREAD

**Issue by montchr** [alleyinteractive/wordpress-fieldmanager#548]
I have a repeating sortable field group containing a set of radio fields. When I drag this group to re-order, the field values sometimes disappear. But when I save the post, the correct values reappear.

![](http://cl.montchr.io/1G2h1v113X45/Screen%20Recording%202016-10-12%20at%2016.41.gif)

**Comment by dlh01** [comment]
I was able to replicate this behavior, and so far I think I've traced it to the call to `fm_renumber()` that occurs in the sortable `stop()` callback: https://github.com/alleyinteractive/wordpress-fieldmanager/blob/master/js/fieldmanager.js#L19.

It's possible for two groups of radio buttons to have the same `name` attribute while `fm_renumber()` is in the middle of renumbering. Only one radio with a given `name` can be checked, so when the naming collision occurs, the later radio becomes unchecked. (If I'm reading correctly, [WordPress core had to fix a similar problem](https://core.trac.wordpress.org/changeset/35809), and I'm extrapolating somewhat from the core bug.)

If this diagnosis is correct, I'm not sure where the correct location in the chain would be to fix it. As a quick starting point, this diff would put the fix in `init_sortable_container()`:

```diff
stop: function( e, ui ) {
	var $parent = ui.item.parents( '.fm-wrapper' ).first();
+	var checked_radios = $parent.find( 'input[type=radio]:checked' );
 	fm_renumber( $parent );
+	$parent.find( checked_radios ).prop( 'checked', true );
 	$( document ).trigger( 'fm_sortable_drop', el );
 }
 ```
